packages/std/src/math/fraction.rs

```diff
@@ -12,3 +12,57 @@ pub trait Fraction<T>: Sized {
     /// If `p` is zero, None is returned.
     fn inv(&self) -> Option<Self>;
 }
+
+impl<T: Clone> Fraction<T> for (T, T) {
+    fn numerator(&self) -> T {
+        self.0.clone()
+    }
+
+    fn denominator(&self) -> T {
+        self.1.clone()
+    }
+
+    fn inv(&self) -> Option<Self> {
+        Some((self.1.clone(), self.0.clone()))
```

Comment:
The inverse needs to fail (return None) then self.0 is 0. Then self is 0/b which makes it a non-interable rationale number.

Reply:
My first thought is to add `Into<Uint256>` which would then allow for a `Uint256::zero()` comparison. Not sure how else to do this though 🤔

Reply:
Oh, good point, there is no obvious way to do `== 0` or `is_zero()` on a generic `T`. Let me think about it ...

Reply:
A slightly simpler way it to make `T: Copy + From<u8>`. Then you can check:

```rust
if self.numerator() == 0u8.into() {
```

We use this in `impl<I> Isqrt for I` already. All relevant typed implement From<u8>, even Uint512. It also removed the Uint256 import in the Fraction definition file.